Enable building uProxy as a Cordova Chrome App for Android

bemasc opened the pull request:
Currently the getter functionality is present but nonfunctional,
so the app is titled "uProxy for Sharers".  The Google social network
is also not working, possibly due to STARTTLS issues in the TCP
socket API bindings.

This is the first part of a solution to https://github.com/uProxy/uproxy/issues/586

<!-- Reviewable:start -->
[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/1809)
<!-- Reviewable:end -->

lucyhe commented:
Looks good! Few small things.

---

Reviewed 8 of 9 files at r1.
Review status: 7 of 9 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

<sup>**[src/cca/app/scripts/cordova_browser_api.ts, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNmk4fvXP3yHRpbtmO)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_browser_api.ts#L57)):</sup>
This comment can probably be removed (made more sense in Chrome where I had a lot more helpers in different categories). 

---

<sup>**[src/cca/app/scripts/cordova_browser_api.ts, line 83 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNnnZNaFN9QNnn_LK_)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_browser_api.ts#L83)):</sup>
Quick note following our chat offline: if we were to assume that tapping the mobile app icon mimics a desktop browserAction (clicking the ext icon), then this code would not work in the case where the user has a uProxy (mobile) window open but out of focus.

---

<sup>**[src/cca/app/scripts/cordova_core_connector.ts, line 23 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNthAeEHvUCX8DKC7x)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_core_connector.ts#L23)):</sup>
We don't need this anymore, if you could take it out. Thanks!

---

<sup>**[src/cca/app/scripts/cordova_core_connector.ts, line 64 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOWlbrJrgQ09nK4O-9)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_core_connector.ts#L64)):</sup>
So I understand, does your making uProxyAppChannel a promise get rid of the need for flushQueue?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->

bemasc commented:
Review status: 6 of 10 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[src/cca/app/scripts/cordova_browser_api.ts, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNmk4fvXP3yHRpbtmO)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_browser_api.ts#L57)):</sup>
Done.

---

<sup>**[src/cca/app/scripts/cordova_browser_api.ts, line 83 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNnnZNaFN9QNnn_LK_)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_browser_api.ts#L83)):</sup>
True.  I've attempted to clarify this code a little.

---

<sup>**[src/cca/app/scripts/cordova_core_connector.ts, line 22 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwNthAeEHvUCX8DKC7x)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_core_connector.ts#L22)):</sup>
Done.

---

<sup>**[src/cca/app/scripts/cordova_core_connector.ts, line 62 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOWlbrJrgQ09nK4O-9)** ([raw file](https://github.com/uproxy/uproxy/blob/d69fc03eb1e99fd0da98111c03a7c2418224295d/src/cca/app/scripts/cordova_core_connector.ts#L62)):</sup>
Sort of.  It means that messages will automatically be queued until the uProxyAppChannel is ready.  It's also important that the channel will never go away once it's present.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->

lucyhe commented:
Review status: 6 of 10 files reviewed at latest revision, 6 unresolved discussions, all commit checks successful.

---

<sup>**[Gruntfile.coffee, line 83 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOijgPvEVZSKK9XX-M)** ([raw file](https://github.com/uproxy/uproxy/blob/d75f080faaa14a29145d39b856baf30393b34e92/Gruntfile.coffee#L83)):</sup>
Why do we need to copy cca twice?

---

<sup>**[Gruntfile.coffee, line 624 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOk8VymdNHN7CAou38)** ([raw file](https://github.com/uproxy/uproxy/blob/d75f080faaa14a29145d39b856baf30393b34e92/Gruntfile.coffee#L624)):</sup>
We have icons, and _locales copied twice.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->

bemasc commented:
Review status: 6 of 10 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[Gruntfile.coffee, line 83 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOijgPvEVZSKK9XX-M)** ([raw file](https://github.com/uproxy/uproxy/blob/d75f080faaa14a29145d39b856baf30393b34e92/Gruntfile.coffee#L83)):</sup>
Removed!

---

<sup>**[Gruntfile.coffee, line 624 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwOk8VymdNHN7CAou38)** ([raw file](https://github.com/uproxy/uproxy/blob/d75f080faaa14a29145d39b856baf30393b34e92/Gruntfile.coffee#L624)):</sup>
Fixed!

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->

lucyhe commented:
last thing then :+1: !

---

Review status: 6 of 10 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

<sup>**[Gruntfile.coffee, line 585 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwPadJLXaB68Oi-e6aw)** ([raw file](https://github.com/uproxy/uproxy/blob/7bc16d6f8391bcbe354271d63d9be9e3bc8133bb/Gruntfile.coffee#L585)):</sup>
!**/*spec* also duplicated below

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->

bemasc commented:
Review status: 6 of 10 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

<sup>**[Gruntfile.coffee, line 585 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/1809#-JwPadJLXaB68Oi-e6aw)** ([raw file](https://github.com/uproxy/uproxy/blob/7bc16d6f8391bcbe354271d63d9be9e3bc8133bb/Gruntfile.coffee#L585)):</sup>
Removed

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/1809)
<!-- Sent from Reviewable.io -->
